core/src/main/java/io/grpc/internal/AbstractChannelBuilder.java

```diff
@@ -45,7 +48,8 @@
  *
  * @param <BuilderT> The concrete type of this builder.
  */
-public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>> {
+public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>>
+    extends ChannelBuilder<BuilderT> {
```

Comment:
The naming would be less confusing if `ChannelBuilder` were an interface instead of an abstract class.

But I like your idea of just getting rid of it. I don't see any material downside to `AndroidChannelBuilder` just extending `Object` and not implementing any interface while providing the same operations as `AbstractChannelBuilder`. So why would removing it just be "for the moment"?